### PR TITLE
Use relative urls where possible and fix broken and outdated links

### PR DESCRIPTION
**Pull request by 36degrees**
_Help me, I've fallen down a link-fixing rabbit hole_ 🐰 🕳️ 

This PR:

- switches to use relative URLs where possible, so that we don't push anyone running the site locally on a preview branch to the live site without them realising (and it makes it easier to maintain)
- tries to appease the link checker (which is currently [not running on CI](https://github.com/alphagov/gds-way/blob/af6200bc43f2fcc1d7add824db8878d33bf732fc/.github/workflows/test.yaml#L33-L35)) by using https wherever possible (looking at you, [yum](http://yum.baseurl.org/wiki/Guides.html) 👀 ) and fixing a few broken links
- removes other outdated references to cloudapps.digital (including updating links to the AWS request an account tool)

### CODE REVIEW

File: source/accessibility.html.md.erb [hunk 1]
```diff
@@ -7,7 +7,7 @@ hide_in_navigation: true
 
 # Accessibility
 
-This accessibility statement applies to the GDS Way at [https://gds-way.digital.cabinet-office.gov.uk/](https://gds-way.digital.cabinet-office.gov.uk/).
+This accessibility statement applies to the GDS Way at <https://gds-way.digital.cabinet-office.gov.uk/>.
```
Comment (lfdebrux) [inline source/accessibility.html.md.erb:10]: Ooh, nice, I didn't know this was a thing!

**Review by lfdebrux — APPROVED** [review]
This is really really awesome, thank you for doing this!

**Comment by lfdebrux** [comment]
Outside of the manual sources there's a few more instances of HTTPS urls that could be fixed; I tried to push a commit myself but I don't have permissions, but here's a patch:

```
From 3d0fbaf079ed94e2a5eddac5a6b68253db38cace Mon Sep 17 00:00:00 2001
From: Laurence de Bruxelles
 <laurence.debruxelles@digital.cabinet-office.gov.uk>
Date: Tue, 23 Apr 2024 09:50:57 +0300
Subject: [PATCH] Use https for URLs where possible

---
 README.md                                         | 6 +++---
 script/deploy                                     | 2 +-
 source/manuals/programming-languages/editorconfig | 2 +-
 3 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/README.md b/README.md
index fdffba5..a4a4c75 100644
--- a/README.md
+++ b/README.md
@@ -161,7 +161,7 @@ The documentation is [© Crown copyright][copyright] and available under the ter
 of the [Open Government 3.0][ogl] licence.
 
 [rvm]: https://www.ruby-lang.org/en/documentation/installation/#managers
-[bundler]: http://bundler.io/
+[bundler]: https://bundler.io/
 [mit]: LICENCE
-[copyright]: http://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/
-[ogl]: http://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/
+[copyright]: https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/
+[ogl]: https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/
diff --git a/script/deploy b/script/deploy
index 758857c..8733bfb 100644
--- a/script/deploy
+++ b/script/deploy
@@ -13,7 +13,7 @@ if [[ ! -f Staticfile.auth ]]; then
   error "Please add a Staticfile.auth file to this directory and try again."
   error
   error "For information on generating a Staticfile.auth, see"
-  error "http://docs.cloudfoundry.org/buildpacks/staticfile/index.html#basic-auth"
+  error "https://docs.cloudfoundry.org/buildpacks/staticfile/index.html#basic-auth"
   exit 1
 fi
 echo "OK!"
diff --git a/source/manuals/programming-languages/editorconfig b/source/manuals/programming-languages/editorconfig
index 320f5bf..5e9c59b 100644
--- a/source/manuals/programming-languages/editorconfig
+++ b/source/manuals/programming-languages/editorconfig
@@ -1,4 +1,4 @@
-# http://EditorConfig.org
+# https://EditorConfig.org
 # https://gds-way.digital.cabinet-office.gov.uk/manuals/programming-languages/editorconfig
 # Copy into project root and rename to .editorconfig
 
-- 
2.44.0
```

File: source/accessibility.html.md.erb (hunk 1 above)
Comment (huwd) [reply source/accessibility.html.md.erb:10]: Huh, TIL ([example from markdown guide](https://www.markdownguide.org/basic-syntax/#urls-and-email-addresses)), cool

File: check_links.rb [hunk 2]
```diff
@@ -8,7 +8,6 @@
     .split("\0")
     .filter { |path| path.start_with? "source/" }
     .map { |path| path.delete_prefix("source/").delete_suffix(".erb").delete_suffix(".md") }
-    .map { |path| "https://gds-way.cloudapps.digital/#{path}" }
 
   proofer = HTMLProofer.check_directory(
```
Comment (huwd) [inline check_links.rb:12]: Great catch, was wondering why the previous changes hadn't been picked up

**Review by huwd — APPROVED** [review]

**Pull request merged**